lib/Doctrine/Common/Annotations/AnnotationRegistry.php

```diff
@@ -142,10 +150,13 @@ static public function loadAnnotationClass($class)
         }
 
         foreach (self::$loaders AS $loader) {
-            if (call_user_func($loader, $class) === true) {
+            if ($loader($class) === true) {
                 return true;
             }
         }
+
+        self::$failedToAutoload[$class] = null;
```

Comment:
Any reason for setting the index instead pushing a new item to the array? It would never be duplicated anyway (since the check is done at the top).

Of course this is a micro optimisation but it might be worth it (also an array with only `null` values is a bit weird IMO)